src/main/java/com/documaster/rms/noark/ws/client/changelog/ChangeLogResult.java

```diff
@@ -0,0 +1,124 @@
+package com.documaster.rms.noark.ws.client.changelog;
+
+import java.time.ZonedDateTime;
+import java.util.Map;
+
+import com.documaster.rms.noark.ws.serialization.ZonedDateTimeDeserializer;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+public class ChangeLogResult {
+
+	private final String id;
+	private final String type;
+	private final String revisionId;
+	private final String revisionType;
+	@JsonDeserialize(using = ZonedDateTimeDeserializer.class)
+	private final ZonedDateTime modifiedDate;
+	private final String modifiedBy;
+	private final String modifiedField;
+	private final String modifiedFieldType;
+
+	private final Map<String, Object> newValue;
+	private final Map<String, Object> oldValue;
+	private final Map<String, Object> removedValue;
+	private final Map<String, Object> previousParent;
+	private final Map<String, Object> currentParent;
+
+	@JsonCreator
+	public ChangeLogResult(
+			@JsonProperty("id") String id,
+			@JsonProperty("type") String type,
+			@JsonProperty("revisionId") String revisionId,
+			@JsonProperty("revisionType") String revisionType,
+			@JsonProperty("modifiedDate") ZonedDateTime modifiedDate,
+			@JsonProperty("modifiedBy") String modifiedBy,
+			@JsonProperty("modifiedField") String modifiedField,
```

Comment:
@ivaylomitrev @dkumanov From here below these are optionally populated based on the revision type. I chose to bundle them in this class, but do we want special deserialization into separate concrete classes for each revision type?

Reply:
I think this is ok for now. We can always create add different deserialization methods in a backwards-compatible way.